pkg/server/metrics/metrics.go

```diff
@@ -187,3 +215,17 @@ func (a *ServerMetrics) SetPendingDialCount(count int) {
 func (a *ServerMetrics) FullRecvChannel(serviceMethod string) prometheus.Gauge {
 	return a.fullRecvChannels.With(prometheus.Labels{"service_method": serviceMethod})
 }
+
+type DialFailureReason string
+
+const (
+	DialFailureEndpoint             DialFailureReason = "endpoint"              // Dial failure reported by the agent back to the server.
```

Comment:
nit: `DialFailureAgent` or `DialFailureBackend` might be slightly clearer IMO

Reply:
Or `DialFailureResponseFromAgent`?

Reply:
How about `ErrorResponse`? Since it's literally a `DIAL_RSP` with an error in it? I can include `...FromAgent` if you prefer.

Reply:
+1 to `DialResponseError` or similar.